### PR TITLE
Syndicate Mail Courier Suit

### PR DESCRIPTION
**Pull request by jorgun**
## About the PR

added new item the Syndicate Mail Courier Suit

## Why / Balance

i cant add a ventcrwaling system, this is the next best thing i can manage.
altso we have mailsystem on more stations
and cargo techs dont have any job uniqe items.

## Media

image
image
image

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- add: added new uniform for syndicate mailman

### CODE REVIEW

**Comment by deltanedas** [comment]
you can literally do this for free with some tools as any role, pipe tagger and regular disposal unit

File: Resources/Prototypes/_Goobstation/Catalog/uplink_catalog.yml [hunk 1]
```diff
@@ -251,6 +251,20 @@
...
+  icon: { sprite: _Goobstation/Clothing/Uniforms/Jumpsuit/mailman.rsi, state: icon }
+  productEntity: ClothingUniformSyndicateMailCarrier
+  cost:
+    Telecrystal: 5
+  categories:
+  - UplinkJob
+  conditions:
+  - !type:BuyerJobCondition
+    whitelist:
+    - CargoTechnician
```
[the remark this answers is not in the document]
Comment (jorgun) [reply Resources/Prototypes/_Goobstation/Catalog/uplink_catalog.yml:267]: that is not the right syntaxt

File: Resources/Prototypes/_Goobstation/Catalog/uplink_catalog.yml [hunk 2]
```diff
@@ -251,6 +251,20 @@
...
     - Passenger
 
+- type: listing
+  id: UplinkSyndicateMailcarrier
+  name: uplink-Syndicate-mail-Carrier-name
+  description: uplink-Syndicate-mail-Carrier-desc
+  icon: { sprite: _Goobstation/Clothing/Uniforms/Jumpsuit/mailman.rsi, state: icon }
+  productEntity: ClothingUniformSyndicateMailCarrier
+  cost:
+    Telecrystal: 5
```
[the remark this answers is not in the document]
Comment (jorgun) [reply Resources/Prototypes/_Goobstation/Catalog/uplink_catalog.yml:261]: maybe, good sugjestion

File: Resources/Prototypes/Entities/Structures/Piping/Disposal/units.yml [hunk 3]
```diff
@@ -120,6 +120,7 @@
     whitelist:
       components:
       - Item
+      - Mailable #Goobstation - Syndicate main uniform
```
[the remark this answers is not in the document]
Comment (jorgun) [reply Resources/Prototypes/Entities/Structures/Piping/Disposal/units.yml:123]: thanks this is a good sugjestion

File: Resources/Locale/en-US/_Goobstation/store/uplink-catalog.ftl [hunk 4]
```diff
@@ -67,3 +67,5 @@ uplink-mech-teleporter-heavy-desc = Contains a heavily armored Cybersun mech wit
 
 uplink-mech-teleporter-assault-name = Assault Mech teleporter
 uplink-mech-teleporter-assault-desc = Contains a lightly armored Cybersun mech with an integrated chainsword, LBX AC 10 "Scattershot", SRM-8 Light Missile Rack and P-X Tesla Cannon.
+uplink-Syndicate-mail-Carrier-name = Syndicate Mail Courier Suit
+uplink-Syndicate-mail-Carrier-desc = Use the mail system as a means of transportation for easy access to various key areas (security, bridge, etc).
```
[the remark this answers is not in the document]
Comment (jorgun) [reply Resources/Locale/en-US/_Goobstation/store/uplink-catalog.ftl:71]: this is a good sugestion

File: Resources/Prototypes/_Goobstation/Entities/Clothing/Uniforms/jumpsuits.yml [hunk 5]
```diff
@@ -80,3 +80,11 @@
...
   - type: Clothing
     sprite: _Goobstation/Clothing/Uniforms/Jumpsuit/hecu.rsi
+
+- type: entity
+  parent: ClothingUniformMailCarrier
+  id: ClothingUniformSyndicateMailCarrier
+  name: syndicate mail carrier's uniform
+  description: You are the mail
+  components:
+  - type: Mailable
```
[the remark this answers is not in the document]
Comment (jorgun) [reply Resources/Prototypes/_Goobstation/Entities/Clothing/Uniforms/jumpsuits.yml:90]: dont need this. we get a sprite from the parent

File: Resources/Prototypes/_Goobstation/Entities/Clothing/Uniforms/jumpsuits.yml [hunk 6]
```diff
@@ -80,3 +80,11 @@
     sprite: _Goobstation/Clothing/Uniforms/Jumpsuit/hecu.rsi
   - type: Clothing
     sprite: _Goobstation/Clothing/Uniforms/Jumpsuit/hecu.rsi
+
+- type: entity
+  parent: ClothingUniformMailCarrier
+  id: ClothingUniformSyndicateMailCarrier
+  name: syndicate mail carrier's uniform
+  description: You are the mail
```
[the remark this answers is not in the document]
Comment (jorgun) [reply Resources/Prototypes/_Goobstation/Entities/Clothing/Uniforms/jumpsuits.yml:88]: this is a bit long for description, can you make it shorter?

File: Content.Server/_Goobstation/Mail/MailableSystem.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+{
+    public sealed class  MailableSystem : EntitySystem{
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<MailableComponent, ClothingGotEquippedEvent>(OnEquipped);
+            SubscribeLocalEvent<MailableComponent, ClothingGotUnequippedEvent>(OnUnequipped);
+        }
```
[the remark this answers is not in the document]
Comment (jorgun) [reply Content.Server/_Goobstation/Mail/MailableSystem.cs:14]: dont need debug here

File: Content.Server/_Goobstation/Mail/MailableSystem.cs [hunk 8]
```diff
@@ -0,0 +1,25 @@
+using Content.Shared.Clothing;
+using Robust.Shared.Utility;
+
+namespace Content.Server._Goobstation.Mailable
+{
+    public sealed class  MailableSystem : EntitySystem{
```
[the remark this answers is not in the document]
Comment (jorgun) [reply Content.Server/_Goobstation/Mail/MailableSystem.cs:6]: this is a good sugestion

File: Content.Server/_Goobstation/Mail/MailableComponent.cs [hunk 9]
```diff
@@ -0,0 +1,7 @@
+namespace Content.Server._Goobstation.Mailable;
+
+[RegisterComponent]
+public sealed partial class MailableComponent : Component
+{
+
+}
```
[the remark this answers is not in the document]
Comment (jorgun) [reply Content.Server/_Goobstation/Mail/MailableComponent.cs:7]: this is a good sugestion

**Comment by Aidenkrz** [comment]
yeah like delta said, kinda useless, also is there literally a ClothingGrantComponent component already?

**Comment by jorgun** [comment]
@Aidenkrz Setting up a disposal system require more knowledge about the system. And if you want to get into sey bridge. There might not be a disposal to get back out.
I was not aware of the clothingAddcComp it whud have been better.

**Comment by deltanedas** [comment]
> There might not be a disposal to get back out.

if you have tools you can convert the mailing unit to disposal and use it

**Pull request closed without merging**